pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -458,20 +467,23 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
 
                                 return null;
                             });
+                            topicLoadRequestSemaphore.get().release();
```

Comment:
The `acquire()` and `release()` operations might not happen on the same semaphore if the limit is being changed.

You should save the instance into a local variable and call release on the same object.